SocialCareCaseViewerApi.Tests/V1/Boundary/Request/UpdateWorkerRequestTests.cs

```diff
@@ -0,0 +1,83 @@
+using NUnit.Framework;
+using SocialCareCaseViewerApi.Tests.V1.Helpers;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+
+namespace SocialCareCaseViewerApi.Tests.V1.Boundary.Request
+{
+    [TestFixture]
+    public class UpdateWorkerRequestTests
+    {
+        private Faker _faker;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _faker = new Faker();
+        }
+
+        [Test]
+        public void UpdateWorkerReturnsIsNotValidForInvalidRequest()
+        {
+            const string longEmail = "thisEmailIsLongerThan62CharactersAndAlsoValid@HereIAmJustCreatingMoreCharacters.com";
+
+            var badRequests = new List<(UpdateWorkerRequest, string)>
+            {
+                (TestHelpers.CreateUpdateWorkersRequest(workerId: 0), "Worker id must be grater than 0"),
+                (TestHelpers.CreateUpdateWorkersRequest(modifiedBy: ""), "Modified by email address must be valid"),
+                (TestHelpers.CreateUpdateWorkersRequest(modifiedBy: longEmail), "Modified by email address must be no longer than 62 characters"),
+                (TestHelpers.CreateUpdateWorkersRequest(email: ""), "Email address must be valid"),
+                (TestHelpers.CreateUpdateWorkersRequest(email: longEmail), "Email address must be no longer than 62 characters"),
+                (TestHelpers.CreateUpdateWorkersRequest(firstName: ""), "First name must be provided"),
+                (TestHelpers.CreateUpdateWorkersRequest(firstName: _faker.Random.String2(101)), "First name must be no longer than 100 characters"),
+                (TestHelpers.CreateUpdateWorkersRequest(lastName: ""), "Last name must be provided"),
+                (TestHelpers.CreateUpdateWorkersRequest(lastName: _faker.Random.String2(101)), "Last name must be no longer than 100 characters"),
+                (TestHelpers.CreateUpdateWorkersRequest(contextFlag: ""), "Context flag must be provided\nContext flag must be 'A' or 'C'"),
+                (TestHelpers.CreateUpdateWorkersRequest(contextFlag: _faker.Random.String2(2, "AC")), "Context flag must be no longer than 1 character"),
+                (TestHelpers.CreateUpdateWorkersRequest(contextFlag: "B"), "Context flag must be 'A' or 'C'"),
+                (TestHelpers.CreateUpdateWorkersRequest(role: ""), "Role must be provided"),
+                (TestHelpers.CreateUpdateWorkersRequest(role: _faker.Random.String2(201)), "Role provided is too long (more than 200 characters)"),
+                (TestHelpers.CreateUpdateWorkersRequest(dateStart: DateTime.Now.AddMinutes(1)), "Date cannot be set in the future"),
+                (TestHelpers.CreateUpdateWorkersRequest(createATeam: false), "A team must be provided"),
+                (TestHelpers.CreateUpdateWorkersRequest(teamName: _faker.Random.String2(201)), "Team name must be no more than 200 characters"),
+                (TestHelpers.CreateUpdateWorkersRequest(teamId: 0), "Team ID must be greater than 0")
+            };
```

Comment:
Another way of doing this is to use `TestCase`, e.g. https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/0c6ea7367fda654f2da94ce451b5b16bac3e1703/SocialCareCaseViewerApi.Tests/V1/Boundary/Request/ListTeamsRequestTests.cs#L52-L61.

Reply:
I think [TestCase()] only works with the primitive data types and not objects.

I could use [TestCaseSource](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcasesource.html) I think, I'm happy to implement this if you think it is tidier.

Reply:
Ahhhhh, that's a shame! Never mind - it's okay as is.